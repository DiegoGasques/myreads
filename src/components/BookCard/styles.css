.book-card {
  min-width: 340px;
}
