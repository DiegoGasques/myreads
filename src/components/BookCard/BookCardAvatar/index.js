import React from "react";
import PropTypes from "prop-types";

import "./styles.css";

function BookCardAvatar({ src }) {
  return (
    <div
      className="book-card-avatar"
      style={{
        backgroundImage: `url(${src})`
      }}
    />
  );
}

BookCardAvatar.defaultProps = {
  src: "https://via.placeholder.com/100x170"
};

BookCardAvatar.propTypes = {
  src: PropTypes.string
};

export default BookCardAvatar;
