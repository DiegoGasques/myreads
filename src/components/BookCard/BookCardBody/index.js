import React from "react";
import PropTypes from "prop-types";

import "./styles.css";

function BookCardBody({ title, authors, children }) {
  return (
    <div className="book-card-body">
      <div className="book-title">{title}</div>
      <div className="book-authors">by {authors.join(", ")}</div>
      {children}
    </div>
  );
}

BookCardBody.defaultProps = {
  title: "Lorem Ipsom Dolour",
  authors: ["Unknown"]
};

BookCardBody.propTypes = {
  title: PropTypes.string.isRequired,
  authors: PropTypes.arrayOf(PropTypes.string)
};

export default BookCardBody;
