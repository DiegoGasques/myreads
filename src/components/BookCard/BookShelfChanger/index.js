import React from "react";
import PropTypes from "prop-types";

import "./styles.css";

function BookShelfChanger({ handleUpdate, currValue }) {
  return (
    <div className="book-shelf-changer-2">
      <select value={currValue} onChange={e => handleUpdate(e.target.value)}>
        <option value="move" disabled>
          Move to...
        </option>
        <option value="currentlyReading">Currently Reading</option>
        <option value="wantToRead">Want to Read</option>
        <option value="read">Read</option>
        <option value="none">None</option>
      </select>
    </div>
  );
}

BookShelfChanger.propTypes = {
  handleUpdate: PropTypes.func.isRequired
};

export default BookShelfChanger;
