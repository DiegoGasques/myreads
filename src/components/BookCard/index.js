import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";

import Card from "../Card";
import BookCardAvatar from "./BookCardAvatar";
import BookCardBody from "./BookCardBody";
import BookShelfChanger from "./BookShelfChanger";
import Ratings from "../StarRatings";

import "./styles.css";

function BookCard({ book, handleUpdate, className }) {
  return (
    <div className={classNames("book-card", className)}>
      <Card>
        <BookCardAvatar
          src={
            book.imageLinks
              ? book.imageLinks.thumbnail
              : "https://via.placeholder.com/100x170"
          }
        />
        <BookCardBody title={book.title} authors={book.authors}>
          <Ratings avgRating={book.averageRating} />
          <BookShelfChanger
            currValue={book.shelf || "none"}
            handleUpdate={shelf => handleUpdate(book, shelf)}
          />
        </BookCardBody>
      </Card>
    </div>
  );
}

BookCard.propTypes = {
  book: PropTypes.object.isRequired,
  handleUpdate: PropTypes.func.isRequired,
  className: PropTypes.string
};

export default BookCard;
