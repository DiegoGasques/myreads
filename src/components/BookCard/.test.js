import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import BookCard from "./index";
import { books } from "../../testFixtures";

it("Should match the component snapshot", () => {
  const handleUpdateSpy = jest.fn();
  const wrapper = shallow(
    <BookCard
      book={books[0]}
      handleUpdate={handleUpdateSpy}
      className="dark-theme"
    />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should call handleUpdate with book and shelf value", () => {
  const handleUpdateSpy = jest.fn();
  const value = "currentlyReading";
  const wrapper = shallow(
    <BookCard
      book={books[2]}
      handleUpdate={handleUpdateSpy}
      className="dark-theme"
    />
  );
  wrapper.find("BookShelfChanger").prop("handleUpdate")(value);
  expect(handleUpdateSpy).toHaveBeenLastCalledWith(books[2], value);
});
