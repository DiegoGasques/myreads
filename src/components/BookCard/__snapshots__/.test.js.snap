// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="book-card dark-theme"
>
  <Card>
    <BookCardAvatar
      src="http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
    />
    <BookCardBody
      authors={
        Array [
          "William E. Shotts, Jr.",
        ]
      }
      title="The Linux Command Line"
    >
      <StarRatings
        avgRating={4}
      />
      <BookShelfChanger
        currValue="currentlyReading"
        handleUpdate={[Function]}
      />
    </BookCardBody>
  </Card>
</div>
`;
